Week 47-48 Release (#90)
aau-hcl-p5/main: report/4.Design/Architecture.tex

As the projectile firing model requires additional calculations, the device will function with merely the laser methods and hardware. 
The methods required for projectile firing are colored red to indicate this.
With the architecture in place, it is now possible to focus on the details of what is required for implementing each part of the architecture.
In the following part, the details regarding the target information structure will be eluded upon.

\subsection{Target information}
A target information structure has been defined internally to store relevant target information.

> This data is calculated on the Raspberry Pi host and the structure then transferred onto the NXT.

Reviewer: ```suggestion
This data is calculated on the Raspberry Pi host and the structure is then transferred onto the NXT.
```

> This data is calculated on the Raspberry Pi host and the structure then transferred onto the NXT.

Reviewer: ```suggestion
This data is calculated on the Raspberry Pi host and the structure then transferred onto the NXT, using the NXT's USB port.
```

The structure can be seen in Listing~\ref{lst:TargetInformation}.
\begin{lstlisting}[language=C, label={lst:TargetInformation},caption={Target information structure from target\_location.h}]
#define TARGET_FOUND 0
#define NO_TARGET_FOUND 1
#define DISCONNECT_REQ 2

typedef struct __attribute__((packed)) {
	int16_t x : 8;
	int16_t y : 8;
} T_TARGET_LOCATION;

typedef struct __attribute__((packed)) {
	T_TARGET_LOCATION location;
	uint16_t status;
} T_TARGET_INFORMATION;
\end{lstlisting}

> The structures uses the packed attribute in order to keep the size at 4 bytes, and 3 status codes are defined with information about whether a target has been found, and whether the host has requested to terminate the connection.

Reviewer: ```suggestion
The structures uses the packed attribute in order to keep the size at 4 bytes, and 3 status codes are defined with information about whether a target has been found, and whether the host has requested termination of the connection.
```
